amp/src/test/java/org/digijava/kernel/ampapi/endpoints/activity/ActivityExporterTest.java

```diff
@@ -0,0 +1,138 @@
+import org.digijava.kernel.ampapi.endpoints.common.TestTranslatorService;
+import org.digijava.kernel.ampapi.endpoints.util.JsonBean;
+import org.digijava.module.aim.dbentity.AmpActivityFields;
+import org.digijava.module.aim.dbentity.AmpActivityProgram;
+import org.digijava.module.aim.dbentity.AmpActivityVersion;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Octavian Ciubotaru
+ */
+public class ActivityExporterTest {
+
+    private List<APIField> fields;
+
+    @Before
+    public void setUp() {
+        TestTranslatorService translatorService = new TestTranslatorService();
+
+        ActivityFieldsEnumerator enumerator = new ActivityFieldsEnumerator(
+                new TestFieldInfoProvider(),
+                new TestFMService(),
+                translatorService,
+                program -> false,
+                "project_code");
+
+        InterchangeUtils.setTranslatorService(translatorService);
+
+        fields = enumerator.getAllAvailableFields(AmpActivityFields.class);
+    }
+
+    @Test
+    public void testProjectName() {
+        AmpActivityVersion activity = new AmpActivityVersion();
+        activity.setName("test");
+
+        Map<String, Object> jsonActivity = createExporter().export(activity).any();
+
+        assertThat(jsonActivity, hasEntry("project_title", "test"));// fixme there are 100+ fields with null values!
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testPrograms() {
+        HardcodedThemes hardcodedThemes = new HardcodedThemes();
+
+        AmpActivityProgram axe1 = new AmpActivityProgram();
+        axe1.setProgramPercentage(100f);
+        axe1.setProgram(hardcodedThemes.getTheme("Axe 1"));
+        axe1.setProgramSetting(hardcodedThemes.getPrimaryProgramSettings());
+
+        AmpActivityProgram instReform = new AmpActivityProgram();
+        instReform.setProgramPercentage(78f);
+        instReform.setProgram(hardcodedThemes.getTheme("Institutional Reform"));
+        instReform.setProgramSetting(hardcodedThemes.getSecondaryProgramSettings());
+
+        AmpActivityProgram edReform = new AmpActivityProgram();
+        edReform.setProgramPercentage(22f);
+        edReform.setProgram(hardcodedThemes.getTheme("Educational Reform"));
+        edReform.setProgramSetting(hardcodedThemes.getSecondaryProgramSettings());
+
+        Set<AmpActivityProgram> programs = new HashSet<>();
+        programs.add(axe1);
+        programs.add(instReform);
+        programs.add(edReform);
+
+        AmpActivityVersion activity = new AmpActivityVersion();
+        activity.setActPrograms(programs);
+
+        Map<String, Object> jsonActivity = createExporter().export(activity).any();
+
+        assertThat(jsonActivity,
+                (Matcher) allOf(
+                        hasEntry(equalTo("primary_programs"),
+                                contains(actProgram(2L, 100F))),
```

Comment:
Since the id is defined in reports tests and if for any reason the id is changed, should it be safer to use theam.getId?